Verify that Strings Are in Valid Email Format: inform about the MailAddress class from start

pkulikov opened the pull request:
- Moved the note about `System.Net.Mail.MailAddress` to the beginning of the topic. 
- Rephrased it such that it should be considered as the main alternative. Should it start with *We recommend...* to make even more stronger?
- Updated the first sentence to relate the whole topic to the set of regex examples.
 
Related to #5305

pkulikov committed:
Inform about MailAddress class at the start

pkulikov committed:
Removed recommendation from the example description

rpetrusha commented on docs/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format.md:
```diff
@@ -22,7 +22,10 @@ author: "rpetrusha"
 ---
 # How to: Verify that Strings Are in Valid Email Format
-The following example uses a regular expression to verify that a string is in valid email format.  
+The following example is more complex than previous examples and uses a regular expression to verify that a string is in valid email format.  
```
This change is problematic. Previous here has no referent. The implicit referent is the TOC, but an extremely small number of people navigate here from the TOC or use the TOC. There's also no clear relationship in terms of complexity between examples. What is the case is that this example is complex. It's OK to phrase this as "The following example uses a complex regular expression..."

rpetrusha replied:
Previous here has no referent. The implicit referent is the TOC, but very few people use it for navigation. The original sentence is succinct and to the point; it's best to leave it as is, or at most to change it to "The following example uses a complex regular expression..."

rpetrusha commented on docs/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format.md:
```diff
@@ -22,7 +22,10 @@ author: "rpetrusha"
+The following example is more complex than previous examples and uses a regular expression to verify that a string is in valid email format.  
+
+> [!NOTE]
+>  Use the <xref:System.Net.Mail.MailAddress?displayProperty=nameWithType> class to check if a string is in valid email address format. To do that, pass the email address string to the <xref:System.Net.Mail.MailAddress.%23ctor%28System.String%29?displayProperty=nameWithType> class constructor that throws a <xref:System.FormatException> if the string has unrecongnized format.  
```
constructor that throws --> constructor, which throws
has --> has an
unrecongnized --> unrecognized

rpetrusha reviewed (changes requested):
Thanks for submitting this PR, @pkulikov. I've left a number of comments.

pkulikov committed:
Addressed feedback

pkulikov commented:
@rpetrusha thank you for the review; I've addressed your feedback. Indeed, my editing of the first sentence was quite weak, I've reverted it to the original version.

rpetrusha reviewed (approved):
Thanks for making the additional set of changes, @pkulikov. I'll merge your PR now.
